WSL/disk-space.md

```diff
@@ -0,0 +1,217 @@
+---
+title: How to manage WSL disk space
+description: Learn how to check the amount of disk space available, expand the size of the Virtual Hard Disk (VHD), repair a VHD mounting or read-only error, and locate the .vhdx file and disk path for Linux distributions installed with WSL 2.
+ms.date: 12/16/2022
+ms.topic: article
+---
+
+# How to manage WSL disk space
+
+This guide covers how to manage the disk space used by Linux distributions installed using WSL 2, including:
+
+- [How to check the amount of disk space available in the VHD](#how-to-check-your-available-disk-space)
+- [How to expand the size of the VHD](#how-to-expand-the-size-of-your-wsl-2-virtual-hard-disk)
+- [How to repair the VHD if an error occurs](#how-to-repair-a-vhd-mounting-error)
+- [How to locate the .vhdx file and disk path for any installed Linux distributions](#how-to-locate-the-vhdx-file-and-disk-path-for-your-linux-distribution)
+
+Windows Subsystem for Linux (WSL 2) uses a virtualization platform to install Linux distributions alongside the host Windows operating system, creating a Virtual Hard Disk (VHD) to store files for each of the Linux distributions that you install. These VHDs use the [ext4 file system type](https://opensource.com/article/17/5/introduction-ext4-filesystem) *(different from the NTFS file system used by Windows)* and are represented on your Windows hard drive as an *ext4.vhdx* file.
+
+WSL 2 automatically resizes these VHD files to meet storage needs. By default each VHD file used by WSL 2 is initially allocated a 1TB maximum amount of disk space (prior to [WSL release 0.58.0](https://github.com/microsoft/WSL/releases/tag/0.58.0) this default was set to a 256GB maximum).
```

Comment:
The default before WSL 0.58.0 was 512GB, not 256GB

Reply:
@OneBlue - Do you know when the switch happened from 256GB to 512GB? May be nice to include that reference as well.

Reply:
This change happened inbox so it's going to be harder to track

Reply:
No worries -- I added a more general statement about it. Thanks for looking.